### ISSUE TITLE
Requesting an NVDA snapshot with new Persian tables embedded.

### ISSUE THREAD

**Issue by mohammadrezarashad** [nvaccess/nvda#6733]
Hello dear friends. Finally I updated the previous Persian literary braille table. I also created a new Persian computer braille table, so Persian users can input Persian characters and symbols using the braille keys on a braille display. I sent them to the Liblouis project. I could fix the problems my tables had with their help and cooperations. And now the tables are ready to use. Now I want to test these tables with NVDA to make sure that they will work properly; especially that computer braille table. I was told that I should compile these tables with NVDA's sourcecode to be able to test them in practice. But since I have no knowledge about that, I want one expert do this for me.
These new tables are now merged to the Liblouis project and probably waiting for next release. But I upload them here for your convenience. I named tables as "fa-ir-g1.utb" for braille output, and "fa-ir-comp8.ctb" for braille input. Thanks in advance

[fa-ir.zip](https://github.com/nvaccess/nvda/files/706218/fa-ir.zip)

**Comment by josephsl** [comment]
Hi, this is contingent on release of Liblouis 3.1.0. I’ll wait until that comes out before returning to this ticket. Thanks.

**Comment by mohammadrezarashad** [comment]
Hi dear Joseph. Thanks for your quick reply. But I want to test them
and fix their probable misbehaviour -especially the input one- before
their official release.

**Comment by josephsl** [comment]
Hmmm, @JCSTeh, @DKager, any thoughts? Thanks.

**Comment by dkager** [comment]
Arguably you should have polished the tables before they were merged into liblouis.
But a quick work-around is to replace an existing table with your new one in the NVDA\louis\tables directory, then selecting that one from the Braille settings.

**Comment by mohammadrezarashad** [comment]
Hi Davy. Surely I examined that. This is not done by just a simple
copy-paste operation.

**Comment by jcsteh** [comment]
It is indeed a simple copy paste if you replace an existing table. You then select the table you replaced. For example, you could replace en-us-g1.ctb with fa-ir-g1.utb and en-us-comp8.ctb with fa-ir-comp8.ctb. To try your tables, you can then select "English (U.S.) grade 1" for output and "English (U.S.) 8 dot computer braille" for input. I suggest you do this in a portable copy of NVDA so you don't break your installed copy.

**Comment by mohammadrezarashad** [comment]
Thanks dear Jamie, I will try it and report the result.

**Comment by mohammadrezarashad** [comment]
Hi. Thanks Dear Jamie, and excuse me dear Davy.
Currently I have no access to a braille display to test braille
output. Of course I borrow one soon. But, using PC Keyboard Braille
Input add-on, I could test my braille input table successfully.
Thanks!

**Comment by mohammadrezarashad** [comment]
Hi dear friends,
Excuse me because of my delay to report. I could finally test my tables with a Seika MiniNotetaker 24-cell braille display a week ago. Of course this coincided with the official release of Liblouis 3.1.0. Fortunately, Persian grade 1 braille output and Persian computer braille tables are OK. Just one problem in My computer braille input:
I assigned dots 3,7,8 and dots 6,7,8 for \x201C and \x201D (leftQuote and rightQuote) characters respectively. There's no problem when I load my computer braille table for braille output; but as braille input, when I key in dots 3,7,8, both with Seika MiniNotetaker braille keys and PC braille input add-on, nothing happens. Dots 6,7,8 inserts rightQuote character of course.
I also have two questions:
1. It seems that when I key in dots 7,8, it just inserts lineFeed character. Is it enough for editing/writing documents in windows applications?
2. I assigned dot 7 for Tab character and dot 8 for \x200C (zero width non-joiner) character. I have no problem using these key assignments when using Seika MiniNotetaker as braille display. But I had to change these for Seika's internal braille input table; Because they assigned these for different actions.
Now my question is, should I change the abovementioned assignments regarding similarity with Seika internal table, or for avoiding probable conflicts with the other braille displays?
I hope my questions are understandable.
     Thanks in advance

**Comment by mohammadrezarashad** [comment]
Any comments or ideas? Still waiting...

**Comment by josephsl** [comment]
Hi, the snapshot you speak up will be made available in June once Liblouis next is released. Thanks.

**Comment by dkager** [comment]
According to https://github.com/nvaccess/nvda/issues/7367#issuecomment-314098057 this table has now been fully tested. Moreover, it will be available in next. Closing this because a try build is no longer needed.
